# Importation la bibliotheque qui nous permetttra de gerer le projet

In [1]:
import pandas as pd

# Importation de notre data set 

In [2]:
df = pd.read_csv("YugiOh - Feuille 1.csv")

In [3]:
df

Nom      Type_1                  Type_2  \
0      Temple Chronomal - Trilithe       Magie                     NaN   
1               1er Mouvement Solo       Magie                     NaN   
2                  À Forces Égales       Piège                     NaN   
3                A Jamais Destinés       Piège                     NaN   
4              À la Recherche d'Or  Compétence        [ Compétence / ]   
...                            ...         ...                     ...   
10831          ZW - Lance Unicorne     Monstre        [ Bête / Effet ]   
10832     ZW - Pégase Double Sabre     Monstre        [ Bête / Effet ]   
10833      ZW - Porteur de Tornade     Monstre      [ Dragon / Effet ]   
10834          ZW - Serres d'Aigle     Monstre  [ Bête-Ailée / Effet ]   
10835           ZW - Vent Sylphide     Monstre        [ Bête / Effet ]   

         Icône                                        Description  Niveau  \
0      Continu  Si vous ne contrôlez aucun monstre, ou si tous...     NaN   
1          NaN  Si vous ne contrôlez aucun monstre : Invoquez ...     NaN   
2          NaN  À la fin de la Battle Phase, si votre adversai...     NaN   
3          NaN  Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN   
4          NaN  Activation de Compétence\n Retournez cette car...     NaN   
...        ...                                                ...     ...   
10831      NaN  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0   
10832      NaN  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0   
10833      NaN  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0   
10834      NaN  Si votre adversaire a min. 2000 Life Points de...     5.0   
10835      NaN  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0   

      Attribut   ATK   DEF  Prix  
0          NaN   NaN   NaN  0,25  
1          NaN   NaN   NaN     6  
2          NaN   NaN   NaN     0  
3          NaN   NaN   NaN  0,25  
4          NaN   NaN   NaN   0,5  
...        ...   ...   ...   ...  
10831  Lumière  1900     0   0,3  
10832  Lumière  1000  2100  0,25  
10833     Vent  1300  1800  0,25  
10834     Vent  2000  1200  0,25  
10835  Lumière   800  1600   0,5  

[10836 rows x 10 columns]

# Info sur le data set 
# Origine du data set : YugiOh est un mangas 
# Circonstances de creation du data set : Ces données ont été collectées après la realisation de l'animé,qfin de mener une campagne publicitaire
# Previsions : On peut faire de previsions sur le prix et le type de carte . En fonction de la description de type de carte on peut donner un prix.
# Aperçu du data: On remarque dans notre data y'a des valeurs manquantes,et aussi que y'a des valeurs qualititatives. Et la dimension matricilelle de notre data set est de 10836 lignes et 10 colonnes. 



# 3 -Affichage des dix premieres lignes de notre data set 

In [4]:
df.head(10)

Nom      Type_1             Type_2       Icône  \
0  Temple Chronomal - Trilithe       Magie                NaN     Continu   
1           1er Mouvement Solo       Magie                NaN         NaN   
2              À Forces Égales       Piège                NaN         NaN   
3            A Jamais Destinés       Piège                NaN         NaN   
4          À la Recherche d'Or  Compétence   [ Compétence / ]         NaN   
5               À Moitié Fermé       Magie                NaN  Jeu Rapide   
6              A.I. Vous défie       Piège                NaN     Continu   
7          A.I. Vous Rencontre       Magie                NaN         NaN   
8                        A.I.Q       Piège                NaN     Continu   
9                        Abaki     Monstre  [ Démon / Effet ]         NaN   

                                         Description  Niveau Attribut   ATK  \
0  Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN   NaN   
1  Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN   NaN   
2  À la fin de la Battle Phase, si votre adversai...     NaN      NaN   NaN   
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN      NaN   NaN   
4  Activation de Compétence\n Retournez cette car...     NaN      NaN   NaN   
5  Ciblez 1 monstre face recto sur le Terrain ;il...     NaN      NaN   NaN   
6  Les monstres contrôlés par votre adversaire pe...     NaN      NaN   NaN   
7  Révélez 1 monstre Cyberse avec 2300 ATK dans v...     NaN      NaN   NaN   
8  Activez uniquement si vous contrôlez un monstr...     NaN      NaN   NaN   
9  Lorsque cette carte est détruite au combat et ...     4.0      Feu  1700   

    DEF  Prix  
0   NaN  0,25  
1   NaN     6  
2   NaN     0  
3   NaN  0,25  
4   NaN   0,5  
5   NaN   0,2  
6   NaN  0,25  
7   NaN  0,25  
8   NaN  0,25  
9  1000     0

# 4- Affichage des 10 dernieres lignes du data set 

In [5]:
df.tail(10)

Nom   Type_1                    Type_2 Icône  \
10826        ZW - Armure Sleipnir  Monstre          [ Bête / Effet ]   NaN   
10827        ZW - Bouclier Ultime  Monstre        [ Rocher / Effet ]   NaN   
10828           ZW - Assaut Asura  Monstre          [ Elfe / Effet ]   NaN   
10829       ZW - Lame Foudroyante  Monstre          [ Bête / Effet ]   NaN   
10830  ZW - Hallebarde Dragonique  Monstre  [ Dragon / Effet / Xyz ]   NaN   
10831         ZW - Lance Unicorne  Monstre          [ Bête / Effet ]   NaN   
10832    ZW - Pégase Double Sabre  Monstre          [ Bête / Effet ]   NaN   
10833     ZW - Porteur de Tornade  Monstre        [ Dragon / Effet ]   NaN   
10834         ZW - Serres d'Aigle  Monstre    [ Bête-Ailée / Effet ]   NaN   
10835          ZW - Vent Sylphide  Monstre          [ Bête / Effet ]   NaN   

                                             Description  Niveau  Attribut  \
10826  Vous pouvez cibler 1 monstre "Utopie" que vous...     4.0   Lumière   
10827  Lorsque cette carte est Invoquée Normalement o...     4.0     Terre   
10828  Vous pouvez cibler 1 monstre "Utopie" que vous...     4.0       Feu   
10829  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0   Lumière   
10830  2 monstres de Niveau 5\n Ne peut pas attaquer ...     5.0  Ténèbres   
10831  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0   Lumière   
10832  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0   Lumière   
10833  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0      Vent   
10834  Si votre adversaire a min. 2000 Life Points de...     5.0      Vent   
10835  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0   Lumière   

        ATK   DEF  Prix  
10826  1000  1000     0  
10827     0  2000  0,25  
10828  1000  1000   0,8  
10829  1200  2000     0  
10830  3000  1200  0,25  
10831  1900     0   0,3  
10832  1000  2100  0,25  
10833  1300  1800  0,25  
10834  2000  1200  0,25  
10835   800  1600   0,5

# 5- Affichage des colonnes de notre data set 

In [6]:
df.columns

Index(['Nom', 'Type_1', 'Type_2', 'Icône', 'Description', 'Niveau', 'Attribut',
       'ATK', 'DEF', 'Prix'],
      dtype='object')

# 6- Mettre le titre de nos colonnes en majiscules

In [7]:
df.columns.str.upper()

Index(['NOM', 'TYPE_1', 'TYPE_2', 'ICÔNE', 'DESCRIPTION', 'NIVEAU', 'ATTRIBUT',
       'ATK', 'DEF', 'PRIX'],
      dtype='object')

# 7- Affichage de la taille du data set 

In [8]:
df.shape

(10836, 10)

# 8-Affichage des informations le type des colonnes 

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10836 entries, 0 to 10835
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Nom          10836 non-null  object 
 1   Type_1       10836 non-null  object 
 2   Type_2       7123 non-null   object 
 3   Icône        1863 non-null   object 
 4   Description  10735 non-null  object 
 5   Niveau       6903 non-null   float64
 6   Attribut     7021 non-null   object 
 7   ATK          6959 non-null   object 
 8   DEF          6959 non-null   object 
 9   Prix         10836 non-null  object 
dtypes: float64(1), object(9)
memory usage: 846.7+ KB


# Affichage des valeurs manquantes  

In [11]:
df.isnull()

Nom  Type_1  Type_2  Icône  Description  Niveau  Attribut    ATK  \
0      False   False    True  False        False    True      True   True   
1      False   False    True   True        False    True      True   True   
2      False   False    True   True        False    True      True   True   
3      False   False    True   True        False    True      True   True   
4      False   False   False   True        False    True      True   True   
...      ...     ...     ...    ...          ...     ...       ...    ...   
10831  False   False   False   True        False   False     False  False   
10832  False   False   False   True        False   False     False  False   
10833  False   False   False   True        False   False     False  False   
10834  False   False   False   True        False   False     False  False   
10835  False   False   False   True        False   False     False  False   

         DEF   Prix  
0       True  False  
1       True  False  
2       True  False  
3       True  False  
4       True  False  
...      ...    ...  
10831  False  False  
10832  False  False  
10833  False  False  
10834  False  False  
10835  False  False  

[10836 rows x 10 columns]

# Affiche toutes la valeurs manquantes de notre dataset 

In [12]:
df.isnull().sum()

Nom               0
Type_1            0
Type_2         3713
Icône          8973
Description     101
Niveau         3933
Attribut       3815
ATK            3877
DEF            3877
Prix              0
dtype: int64

# Oui ce data set admet des valeurs manquantes dans les colonnes ci apres :
# Type_2
# Icone
# Description
# Niveau
# Attribut
# ATK
# DEF


# 9- Affichage des occurences de la colonne "Type_2"

In [13]:
df["Type_2"].value_counts()

[ Guerrier / Effet ]            543
[ Machine / Effet ]             451
[ Démon / Effet ]               365
[ Magicien / Effet ]            332
[ Elfe / Effet ]                280
                               ... 
[ Aqua / Spirit / Effet ]         1
[ Insecte / Fusion / Effet ]      1
[ Effet / Plante ]                1
[ Reptile / Syntoniseur ]         1
[ Dragon / Effet / Xyz ]          1
Name: Type_2, Length: 468, dtype: int64

# 10- Affichage du contenu des 5 premieres lignes de la colonne "Description"

In [17]:
desc = df["Description"]
desc.head(5)

0    Si vous ne contrôlez aucun monstre, ou si tous...
1    Si vous ne contrôlez aucun monstre : Invoquez ...
2    À la fin de la Battle Phase, si votre adversai...
3    Ciblez 2 monstres dans votre Cimetière ; Invoq...
4    Activation de Compétence\n Retournez cette car...
Name: Description, dtype: object

In [19]:
df["Description"].head(5)

0    Si vous ne contrôlez aucun monstre, ou si tous...
1    Si vous ne contrôlez aucun monstre : Invoquez ...
2    À la fin de la Battle Phase, si votre adversai...
3    Ciblez 2 monstres dans votre Cimetière ; Invoq...
4    Activation de Compétence\n Retournez cette car...
Name: Description, dtype: object

# 11- Affichage du contenu des 5 premieres lignes des colonnes , "Description","Type_2" et "Type_1"

In [26]:
desc =df["Description"]
T1 = df["Type_1"]
T2 = df["Type_2"]

df2 = pd.concat([desc,T1,T2], axis=1)

df2.head(5)

Description      Type_1  \
0  Si vous ne contrôlez aucun monstre, ou si tous...       Magie   
1  Si vous ne contrôlez aucun monstre : Invoquez ...       Magie   
2  À la fin de la Battle Phase, si votre adversai...       Piège   
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...       Piège   
4  Activation de Compétence\n Retournez cette car...  Compétence   

             Type_2  
0               NaN  
1               NaN  
2               NaN  
3               NaN  
4  [ Compétence / ]

In [28]:
 pd.concat([df["Description"],df["Type_1"],df["Type_2"]], axis=1).head(5)

Description      Type_1  \
0  Si vous ne contrôlez aucun monstre, ou si tous...       Magie   
1  Si vous ne contrôlez aucun monstre : Invoquez ...       Magie   
2  À la fin de la Battle Phase, si votre adversai...       Piège   
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...       Piège   
4  Activation de Compétence\n Retournez cette car...  Compétence   

             Type_2  
0               NaN  
1               NaN  
2               NaN  
3               NaN  
4  [ Compétence / ]

# 12- Affichage du contenu de la premiere ligne et de la cinquieme colonne 

In [31]:
df.iloc[0,4]

'Si vous ne contrôlez aucun monstre, ou si tous les monstres que vous contrôlez sont des monstres "Chronomal" : vous pouvez payer 500 LP ; immédiatement après la résolution de cet effet, Invoquez Normalement 1 monstre "Chronomal" depuis votre main. Si un Monstre Xyz "Chronomal" ou "Numéro" que vous contrôlez va activer un effet en détachant du Matériel, vous pouvez envoyer cette carte au Cimetière au lieu d\'1 de ces Matériels. Vous ne pouvez utiliser chaque effet de "Temple Chronomal - Trilithe" qu\'une fois par tour.'

In [32]:
desc =df["Description"]

desc[0]

'Si vous ne contrôlez aucun monstre, ou si tous les monstres que vous contrôlez sont des monstres "Chronomal" : vous pouvez payer 500 LP ; immédiatement après la résolution de cet effet, Invoquez Normalement 1 monstre "Chronomal" depuis votre main. Si un Monstre Xyz "Chronomal" ou "Numéro" que vous contrôlez va activer un effet en détachant du Matériel, vous pouvez envoyer cette carte au Cimetière au lieu d\'1 de ces Matériels. Vous ne pouvez utiliser chaque effet de "Temple Chronomal - Trilithe" qu\'une fois par tour.'

# 13 - Affichage des 5 premieres lignes des colonnes 5 a 7

In [34]:
desc =df["Description"]
Niv = df["Niveau"]
Att = df["Attribut"]

df2 = pd.concat([desc,Niv,Att], axis=1)

df2.head(5)

Description  Niveau Attribut
0  Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN
1  Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN
2  À la fin de la Battle Phase, si votre adversai...     NaN      NaN
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN      NaN
4  Activation de Compétence\n Retournez cette car...     NaN      NaN

In [36]:
pd.concat([df["Description"],df["Niveau"],df["Attribut"]], axis =1).head(5)

Description  Niveau Attribut
0  Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN
1  Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN
2  À la fin de la Battle Phase, si votre adversai...     NaN      NaN
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN      NaN
4  Activation de Compétence\n Retournez cette car...     NaN      NaN

In [38]:
df.iloc[0:5,4:7]

Description  Niveau Attribut
0  Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN
1  Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN
2  À la fin de la Battle Phase, si votre adversai...     NaN      NaN
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN      NaN
4  Activation de Compétence\n Retournez cette car...     NaN      NaN

# 14- Copie du data dans une variable avec les valeurs de la colonne type_1 = monstre 

In [40]:
set = df[(df["Type_1"]=="Monstre")]

set

Nom   Type_1                          Type_2  \
9                         Abaki  Monstre               [ Démon / Effet ]   
10                Abare Ushioni  Monstre       [ Bête-Guerrier / Effet ]   
11        Abanc le Mythologique  Monstre  [ Bête / Syntoniseur / Effet ]   
12             Abeille Automate  Monstre             [ Insecte / Effet ]   
13            Abeille Cuirassée  Monstre             [ Insecte / Effet ]   
...                         ...      ...                             ...   
10831       ZW - Lance Unicorne  Monstre                [ Bête / Effet ]   
10832  ZW - Pégase Double Sabre  Monstre                [ Bête / Effet ]   
10833   ZW - Porteur de Tornade  Monstre              [ Dragon / Effet ]   
10834       ZW - Serres d'Aigle  Monstre          [ Bête-Ailée / Effet ]   
10835        ZW - Vent Sylphide  Monstre                [ Bête / Effet ]   

      Icône                                        Description  Niveau  \
9       NaN  Lorsque cette carte est détruite au combat et ...     4.0   
10      NaN  Prenez une pièce et jouez à pile ou face. Si v...     4.0   
11      NaN  Si cette carte est dans votre Cimetière : vous...     2.0   
12      NaN  Lorsque cette carte est détruite au combat et ...     1.0   
13      NaN  Une fois par tour : vous pouvez cibler 1 monst...     4.0   
...     ...                                                ...     ...   
10831   NaN  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0   
10832   NaN  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0   
10833   NaN  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0   
10834   NaN  Si votre adversaire a min. 2000 Life Points de...     5.0   
10835   NaN  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0   

      Attribut   ATK   DEF  Prix  
9          Feu  1700  1000     0  
10       Terre  1200  1200     0  
11     Lumière  1000   600  0,25  
12       Terre   100   300  0,25  
13        Vent  1600  1200     1  
...        ...   ...   ...   ...  
10831  Lumière  1900     0   0,3  
10832  Lumière  1000  2100  0,25  
10833     Vent  1300  1800  0,25  
10834     Vent  2000  1200  0,25  
10835  Lumière   800  1600   0,5  

[7024 rows x 10 columns]

In [41]:
set = df[(df["Type_1"]=="Magie")]

set

Nom Type_1 Type_2       Icône  \
0           Temple Chronomal - Trilithe  Magie    NaN     Continu   
1                    1er Mouvement Solo  Magie    NaN         NaN   
5                        À Moitié Fermé  Magie    NaN  Jeu Rapide   
7                   A.I. Vous Rencontre  Magie    NaN         NaN   
19                 Abondance du Cristal  Magie    NaN         NaN   
...                                 ...    ...    ...         ...   
10803         Zone de Danger Surnaturel  Magie    NaN     Continu   
10804                   Zone de Défense  Magie    NaN     Terrain   
10806  Zone de Neutralisation des Sorts  Magie    NaN     Continu   
10808                   Zone De Progrès  Magie    NaN     Continu   
10809           Zone de Repos Psychique  Magie    NaN         NaN   

                                             Description  Niveau Attribut  \
0      Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN   
1      Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN   
5      Ciblez 1 monstre face recto sur le Terrain ;il...     NaN      NaN   
7      Révélez 1 monstre Cyberse avec 2300 ATK dans v...     NaN      NaN   
19     Envoyez 4 cartes "Bête Cristalline" depuis vot...     NaN      NaN   
...                                                  ...     ...      ...   
10803  Si un monstre (Monstres à Effet exclus) est In...     NaN      NaN   
10804  Pour chaque monstre qu'un joueur contrôle dans...     NaN      NaN   
10806  Tous les monstres face recto sur le Terrain, e...     NaN      NaN   
10808  Une fois par tour, durant la End Phase d'un to...     NaN      NaN   
10809  Sélectionnez 2 de vos monstres retirés du jeu ...     NaN      NaN   

       ATK  DEF  Prix  
0      NaN  NaN  0,25  
1      NaN  NaN     6  
5      NaN  NaN   0,2  
7      NaN  NaN  0,25  
19     NaN  NaN  0,25  
...    ...  ...   ...  
10803  NaN  NaN  0,25  
10804  NaN  NaN  0,25  
10806  NaN  NaN   1,5  
10808  NaN  NaN    11  
10809  NaN  NaN     0  

[2097 rows x 10 columns]

# 15- Affichage des informations sur cette variable 

In [42]:
set.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2097 entries, 0 to 10809
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Nom          2097 non-null   object 
 1   Type_1       2097 non-null   object 
 2   Type_2       0 non-null      object 
 3   Icône        1288 non-null   object 
 4   Description  2096 non-null   object 
 5   Niveau       0 non-null      float64
 6   Attribut     0 non-null      object 
 7   ATK          0 non-null      object 
 8   DEF          0 non-null      object 
 9   Prix         2097 non-null   object 
dtypes: float64(1), object(9)
memory usage: 180.2+ KB


In [43]:
set.info

<bound method DataFrame.info of                                     Nom Type_1 Type_2       Icône  \
0           Temple Chronomal - Trilithe  Magie    NaN     Continu   
1                    1er Mouvement Solo  Magie    NaN         NaN   
5                        À Moitié Fermé  Magie    NaN  Jeu Rapide   
7                   A.I. Vous Rencontre  Magie    NaN         NaN   
19                 Abondance du Cristal  Magie    NaN         NaN   
...                                 ...    ...    ...         ...   
10803         Zone de Danger Surnaturel  Magie    NaN     Continu   
10804                   Zone de Défense  Magie    NaN     Terrain   
10806  Zone de Neutralisation des Sorts  Magie    NaN     Continu   
10808                   Zone De Progrès  Magie    NaN     Continu   
10809           Zone de Repos Psychique  Magie    NaN         NaN   

                                             Description  Niveau Attribut  \
0      Si vous ne contrôlez aucun monstre, ou si tous...     N

# 16 - Affichage du data set en fonction de type_1 = magie et icone = continu

In [51]:
data = df[(df["Type_1"] =="Magie") & (df["Icône"] =="Continu")]

data

Nom Type_1 Type_2    Icône  \
0           Temple Chronomal - Trilithe  Magie    NaN  Continu   
23                  Absorption de l'Âme  Magie    NaN  Continu   
26                  Absorption de Magie  Magie    NaN  Continu   
29             Absorption Super Solaire  Magie    NaN  Continu   
35              Accélérateur de Flammes  Magie    NaN  Continu   
...                                 ...    ...    ...      ...   
10652     Walhalla, Sanctuaire du Déchu  Magie    NaN  Continu   
10773                        Zérozérock  Magie    NaN  Continu   
10803         Zone de Danger Surnaturel  Magie    NaN  Continu   
10806  Zone de Neutralisation des Sorts  Magie    NaN  Continu   
10808                   Zone De Progrès  Magie    NaN  Continu   

                                             Description  Niveau Attribut  \
0      Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN   
23     A chaque fois qu'une carte est retirée du jeu,...     NaN      NaN   
26     A chaque fois qu'une Carte Magie est activée, ...     NaN      NaN   
29     Sacrifiez 1 monstre de Type Plante de Niveau 2...     NaN      NaN   
35     En envoyant 1 monstre de Type Pyro avec une AT...     NaN      NaN   
...                                                  ...     ...      ...   
10652  Une fois par tour, si vous ne contrôlez aucun ...     NaN      NaN   
10773  Les Monstres de votre adversaire ne peuvent pa...     NaN      NaN   
10803  Si un monstre (Monstres à Effet exclus) est In...     NaN      NaN   
10806  Tous les monstres face recto sur le Terrain, e...     NaN      NaN   
10808  Une fois par tour, durant la End Phase d'un to...     NaN      NaN   

       ATK  DEF  Prix  
0      NaN  NaN  0,25  
23     NaN  NaN     3  
26     NaN  NaN     0  
29     NaN  NaN   1,5  
35     NaN  NaN   0,5  
...    ...  ...   ...  
10652  NaN  NaN     0  
10773  NaN  NaN  0,25  
10803  NaN  NaN  0,25  
10806  NaN  NaN   1,5  
10808  NaN  NaN    11  

[362 rows x 10 columns]

# 17- Conversion des colonnes ATK et DEF en entiers et la colonne prix en float 

In [54]:
df["ATK"] =df["ATK"].astype(int)
df["Nom"] = df["Nom"].astype(int)

ValueError: cannot convert float NaN to integer

# Cette syntaxe ne marche pas car il faut d'abord convertir le type object en string avant de le convertir en entiers ou en flottant 

df["Prix"] = df["Prix"].astype(int)

# On peut pas convertir Prix en entier 

# 18 - Suppression des valeurs manquantes dans les colonnes type_2 et ATK

In [60]:
df2 = pd.concat([df["Type_2"], df["ATK"]], axis = 1)

df2.dropna()

Type_2   ATK
9                   [ Démon / Effet ]  1700
10          [ Bête-Guerrier / Effet ]  1200
11     [ Bête / Syntoniseur / Effet ]  1000
12                [ Insecte / Effet ]   100
13                [ Insecte / Effet ]  1600
...                               ...   ...
10831                [ Bête / Effet ]  1900
10832                [ Bête / Effet ]  1000
10833              [ Dragon / Effet ]  1300
10834          [ Bête-Ailée / Effet ]  2000
10835                [ Bête / Effet ]   800

[6959 rows x 2 columns]

In [61]:
pd.concat([df["Type_2"], df["ATK"]], axis = 1).dropna()

Type_2   ATK
9                   [ Démon / Effet ]  1700
10          [ Bête-Guerrier / Effet ]  1200
11     [ Bête / Syntoniseur / Effet ]  1000
12                [ Insecte / Effet ]   100
13                [ Insecte / Effet ]  1600
...                               ...   ...
10831                [ Bête / Effet ]  1900
10832                [ Bête / Effet ]  1000
10833              [ Dragon / Effet ]  1300
10834          [ Bête-Ailée / Effet ]  2000
10835                [ Bête / Effet ]   800

[6959 rows x 2 columns]

In [63]:
dataMonstre = df[(df["Type_1"]=="Monstre")]

dataMonstre

Nom   Type_1                          Type_2  \
9                         Abaki  Monstre               [ Démon / Effet ]   
10                Abare Ushioni  Monstre       [ Bête-Guerrier / Effet ]   
11        Abanc le Mythologique  Monstre  [ Bête / Syntoniseur / Effet ]   
12             Abeille Automate  Monstre             [ Insecte / Effet ]   
13            Abeille Cuirassée  Monstre             [ Insecte / Effet ]   
...                         ...      ...                             ...   
10831       ZW - Lance Unicorne  Monstre                [ Bête / Effet ]   
10832  ZW - Pégase Double Sabre  Monstre                [ Bête / Effet ]   
10833   ZW - Porteur de Tornade  Monstre              [ Dragon / Effet ]   
10834       ZW - Serres d'Aigle  Monstre          [ Bête-Ailée / Effet ]   
10835        ZW - Vent Sylphide  Monstre                [ Bête / Effet ]   

      Icône                                        Description  Niveau  \
9       NaN  Lorsque cette carte est détruite au combat et ...     4.0   
10      NaN  Prenez une pièce et jouez à pile ou face. Si v...     4.0   
11      NaN  Si cette carte est dans votre Cimetière : vous...     2.0   
12      NaN  Lorsque cette carte est détruite au combat et ...     1.0   
13      NaN  Une fois par tour : vous pouvez cibler 1 monst...     4.0   
...     ...                                                ...     ...   
10831   NaN  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0   
10832   NaN  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0   
10833   NaN  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0   
10834   NaN  Si votre adversaire a min. 2000 Life Points de...     5.0   
10835   NaN  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0   

      Attribut   ATK   DEF  Prix  
9          Feu  1700  1000     0  
10       Terre  1200  1200     0  
11     Lumière  1000   600  0,25  
12       Terre   100   300  0,25  
13        Vent  1600  1200     1  
...        ...   ...   ...   ...  
10831  Lumière  1900     0   0,3  
10832  Lumière  1000  2100  0,25  
10833     Vent  1300  1800  0,25  
10834     Vent  2000  1200  0,25  
10835  Lumière   800  1600   0,5  

[7024 rows x 10 columns]

In [64]:
dataMonstre.to_csv('DataYugiOh.csv', index = False)

In [66]:
data = pd.read_csv("DataYugiOh.csv")

data

Nom   Type_1                          Type_2  \
0                        Abaki  Monstre               [ Démon / Effet ]   
1                Abare Ushioni  Monstre       [ Bête-Guerrier / Effet ]   
2        Abanc le Mythologique  Monstre  [ Bête / Syntoniseur / Effet ]   
3             Abeille Automate  Monstre             [ Insecte / Effet ]   
4            Abeille Cuirassée  Monstre             [ Insecte / Effet ]   
...                        ...      ...                             ...   
7019       ZW - Lance Unicorne  Monstre                [ Bête / Effet ]   
7020  ZW - Pégase Double Sabre  Monstre                [ Bête / Effet ]   
7021   ZW - Porteur de Tornade  Monstre              [ Dragon / Effet ]   
7022       ZW - Serres d'Aigle  Monstre          [ Bête-Ailée / Effet ]   
7023        ZW - Vent Sylphide  Monstre                [ Bête / Effet ]   

      Icône                                        Description  Niveau  \
0       NaN  Lorsque cette carte est détruite au combat et ...     4.0   
1       NaN  Prenez une pièce et jouez à pile ou face. Si v...     4.0   
2       NaN  Si cette carte est dans votre Cimetière : vous...     2.0   
3       NaN  Lorsque cette carte est détruite au combat et ...     1.0   
4       NaN  Une fois par tour : vous pouvez cibler 1 monst...     4.0   
...     ...                                                ...     ...   
7019    NaN  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0   
7020    NaN  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0   
7021    NaN  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0   
7022    NaN  Si votre adversaire a min. 2000 Life Points de...     5.0   
7023    NaN  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0   

     Attribut   ATK   DEF  Prix  
0         Feu  1700  1000     0  
1       Terre  1200  1200     0  
2     Lumière  1000   600  0,25  
3       Terre   100   300  0,25  
4        Vent  1600  1200     1  
...       ...   ...   ...   ...  
7019  Lumière  1900     0   0,3  
7020  Lumière  1000  2100  0,25  
7021     Vent  1300  1800  0,25  
7022     Vent  2000  1200  0,25  
7023  Lumière   800  1600   0,5  

[7024 rows x 10 columns]